optimism/src/keccak/trace.rs

```diff
@@ -56,7 +56,26 @@ impl<F: Field> CircuitTrait<ZKVM_KECCAK_COLS, Steps, F, KeccakEnv<F>> for Keccak
         });
     }
 
-    fn pad(&mut self, step: Steps) -> bool {
+    fn pad_with_row(&mut self, step: Steps, row: &[F; ZKVM_KECCAK_COLS]) -> bool {
+        let mut step = step;
+        if let Round(_) = step {
+            step = Round(0);
+        }
+        let rows_left = self.domain_size - self.witness[&step].cols[0].len();
+        if rows_left == 0 {
```

Comment:
Just in case as not explicit for the runtime here: what about `rows_left < 0`.

Reply:
No need because it is a usize 😉

Reply:
You need, even if it is a usize. https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=c6c1c1dafe3cffe462cf019c2216ecdc